tests/python/test_slowmo_fsdp.py

```diff
@@ -0,0 +1,378 @@
+        )
+
+        slowmo_optim = self._get_slomo_optimizer(
+            base_optim=torch.optim.SGD(fsdp_net_slowmo.parameters(), lr=1e-2),
+            slowmo_freq=6,
+        )
+
+        # Manually changing slow momentum optimizer's averager's period
+        # to differ from `slowmo_freq` to check it independently from
+        # the momentum's update. Basically, parameter averaging now will happen
+        # every 3rd step and momentum step every 6th.
+        slowmo_optim.averager.period = 3
+
+        averager2 = self._get_averager(period=3)
+        base_optimizer = torch.optim.SGD(fsdp_net.parameters(), lr=1e-2)
+
+        for _ in range(4):
+            self._train_step(inpt, fsdp_net, base_optimizer)
+            self._train_step(inpt, fsdp_net_slowmo, slowmo_optim)
+            averager2.average_parameters(fsdp_net.parameters())
+
+        for slowmo_params, net_params in zip(
+            fsdp_net_slowmo.parameters(), fsdp_net.parameters()
+        ):
+            self.assertEqual(slowmo_params, net_params)
+
+    @skip_if_lt_x_gpu(2)
+    @parametrize("sharding_strategy", [ShardingStrategy.NO_SHARD])
+    def test_slowmo_optimizer_momentum_step(
+        self, sharding_strategy: Optional[ShardingStrategy]
+    ):
+        fsdp_net = self._init_fsdp(sharding_strategy)
+        fsdp_net_slowmo = self._init_fsdp(sharding_strategy)
+
+        cur_subgroup = dist.new_group(ranks=[self.rank])
+        slowmo_state = slow_momentum_comm.SlowMoState(cur_subgroup)
+        fsdp_net.register_comm_hook(slowmo_state, slow_momentum_comm.slowmo_hook)
+        fsdp_net_slowmo.register_comm_hook(slowmo_state, slow_momentum_comm.slowmo_hook)
+        inpt = torch.tensor(
+            [(self.rank + 1)],
+            dtype=torch.float,
+            device=self.rank,  # type: ignore[arg-type]
+        )
+
+        slowmo_optim = self._get_slomo_optimizer(
+            base_optim=torch.optim.SGD(fsdp_net_slowmo.parameters(), lr=1e-2),
+            slowmo_freq=2,
+        )
+        averager2 = self._get_averager(period=2)
+        base_optimizer = torch.optim.SGD(fsdp_net.parameters(), lr=1e-2)
+
+        for _, v in slowmo_optim.state.items():
+            initial_prev_params = copy.deepcopy(v["prev_param"])
+            initial_slow_momentum_buffer = copy.deepcopy(v["slow_momentum"])
+
+        for _ in range(3):
+            self._train_step(inpt, fsdp_net, base_optimizer)
+            self._train_step(inpt, fsdp_net_slowmo, slowmo_optim)
+            averager2.average_parameters(fsdp_net.parameters())
+
+        # parameters before slow momentum update and after averaging
+        # are in `fsdp_net.params[0]`
+        # can use them to calculate momentum update
+        # momentum_(t+1) = slowmo_factor * momentum_t +
+        #   (prev_param - cur_param)/base_lr
+        momentum = (
+            slowmo_optim.slowmo_factor * initial_slow_momentum_buffer
+            + (initial_prev_params - fsdp_net.params[0]) / 0.01
```

Comment:
Shall we refactor `1e-2` above and `0.01` here into a `LR = 1e-2` variable to make it clearer that they are expected to be the same?

Reply:
good point, will do